pkg/expr/sql/frame_db_conv.go

```diff
@@ -0,0 +1,471 @@
+		}
+		return nil, fmt.Errorf("unexpected value type for interface %v, expected uint64 or *uint64", val)
+
+	// ----------------------------
+	// Float64 / Nullable Float64
+	// ----------------------------
+	case data.FieldTypeFloat64:
+		// Accept float64 or decimal.Decimal, convert decimal.Decimal -> float64
+		if v, ok := val.(float64); ok {
+			return v, nil
+		}
+		if d, ok := val.(decimal.Decimal); ok {
+			return d.InexactFloat64(), nil
+		}
+		return nil, fmt.Errorf("unexpected value type for interface %v, expected float64 or decimal.Decimal", val)
+
+	case data.FieldTypeNullableFloat64:
+		// Possibly already *float64
+		if vP, ok := val.(*float64); ok {
+			return vP, nil
+		}
+		// Possibly float64
+		if v, ok := val.(float64); ok {
+			return &v, nil
+		}
+		// Possibly decimal.Decimal
+		if d, ok := val.(decimal.Decimal); ok {
+			f := d.InexactFloat64()
+			return &f, nil
+		}
+		return nil, fmt.Errorf("unexpected value type for interface %v, expected float64, *float64, or decimal.Decimal", val)
+
+	// ----------------------------
+	// Time / Nullable Time
+	// ----------------------------
+	case data.FieldTypeTime:
+		v, ok := val.(time.Time)
+		if !ok {
+			return nil, fmt.Errorf("unexpected value type for interface %v, expected time.Time", val)
+		}
+		return v, nil
+
+	case data.FieldTypeNullableTime:
+		vP, ok := val.(*time.Time)
+		if ok {
+			return vP, nil
+		}
+		v, ok := val.(time.Time)
+		if ok {
+			return &v, nil
+		}
+		return nil, fmt.Errorf("unexpected value type for interface %v, expected time.Time or *time.Time", val)
+
+	// ----------------------------
+	// String / Nullable String
+	// ----------------------------
+	case data.FieldTypeString:
+		v, ok := val.(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected value type for interface %v, expected string", val)
+		}
+		return v, nil
+
+	case data.FieldTypeNullableString:
+		vP, ok := val.(*string)
+		if ok {
+			return vP, nil
+		}
+		v, ok := val.(string)
+		if ok {
+			return &v, nil
+		}
+		return nil, fmt.Errorf("unexpected value type for interface %v, expected string or *string", val)
+
+	// ----------------------------
+	// Bool / Nullable Bool
+	// ----------------------------
+	case data.FieldTypeBool:
+		v, ok := val.(bool)
+		if !ok {
+			return nil, fmt.Errorf("unexpected value type for interface %v, expected bool", val)
+		}
+		return v, nil
+
+	case data.FieldTypeNullableBool:
+		vP, ok := val.(*bool)
+		if ok {
+			return vP, nil
+		}
+		v, ok := val.(bool)
+		if ok {
+			return &v, nil
+		}
+		return nil, fmt.Errorf("unexpected value type for interface %v, expected bool or *bool", val)
+
+	// ----------------------------
+	// Fallback / Unsupported
+	// ----------------------------
+	default:
+		return nil, fmt.Errorf("unsupported field type %s for val %v", fieldType, val)
+	}
+}
+
+// Is the field nilAt the index. Can panic if out of range.
+// TODO: Maybe this should be a method on data.Field?
+func nilAt(field data.Field, at int) bool {
```

Comment:
This can go once https://github.com/grafana/grafana-plugin-sdk-go/commit/d477c83424db327262e18ed3022be49fd6af2a4d finds its way into an SDK release and into Grafana